### PR TITLE
variable naming convention with screaming case

### PR DESCRIPTION
**Pull request by antspk**
Update variable naming convention to include `SCREAMING_CASE` for global variables and static class properties

closes: #1043

### CODE REVIEW

**Review by elektronik2k5 — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -338,24 +338,32 @@ const count = 2 // it tries to run 2(), but 2 is not a function
...
 ### 3.6 Code Example
 
 ```javascript
-// for class name we use UpperCamelCase
-class SomeClassExample {}
-
-// for const names we use the const keyword and lowerCamelCase
-const config = {
+// for global variables names we use the const/let keyword and SCREAMING_CASE
+let MUTABLE_GLOBAL = "mutable value"
```
Comment (rluvaton) [inline README.md:349]: Why `let` in screaming case?

I though only constants

File: README.md [hunk 2]
```diff
@@ -338,24 +338,32 @@ const count = 2 // it tries to run 2(), but 2 is not a function
 
 ## ![✔] 3.6 Use naming conventions for variables, constants, functions and classes
 
-**TL;DR:** Use **_lowerCamelCase_** when naming constants, variables and functions and **_UpperCamelCase_** (capital first letter as well) when naming classes. This will help you to easily distinguish between plain variables/functions, and classes that require instantiation. Use descriptive names, but try to keep them short
+**TL;DR:** Use **_lowerCamelCase_** when naming constants, variables and functions and **_UpperCamelCase_** (capital first letter as well) when naming classes, **_SCREAMING_CASE_** when naming global or static variables. This will help you to easily distinguish between plain variables/functions, classes that require instantiation and variables declared at global module scope. Use descriptive names, but try to keep them short
```
Comment (rluvaton) [inline README.md:341]: I think we should add somewhere why screaming case 🤔

File: README.md (hunk 1 above)
Comment (elektronik2k5) [reply README.md:349]: Cause it is a global. Personally, I'm ok with this. Globals are bad, we should avoid them, so screaming them seems fit! 🗣️ 📢

File: README.md (hunk 2 above)
Comment (elektronik2k5) [reply README.md:341]: The more conservative name is "upper snake case".

Comment (antspk) [reply README.md]: > I think we should add somewhere why screaming case 🤔

I've changed convention name to the upper snake case. 

Regarding statement why, I think this statement explains it: 
`This will help you to easily distinguish ...  variables declared at global module scope`.
Or does it need separation from the rest?

Comment (rluvaton) [reply README.md]: What I meant is maybe show example of this convention is used in Node (like PI as you mentioned in the issue) but on a second thought I think it should be here but rather in the code example

File: README.md [hunk 3]
```diff
@@ -338,24 +338,32 @@ const count = 2 // it tries to run 2(), but 2 is not a function
...
 
 ### 3.6 Code Example
 
 ```javascript
-// for class name we use UpperCamelCase
-class SomeClassExample {}
-
-// for const names we use the const keyword and lowerCamelCase
-const config = {
+// for global variables names we use the const/let keyword and UPPER_SNAKE_CASE
```
Comment (rluvaton) [inline README.md:348]: Maybe add here - _(like `Math.PI`)_

Comment (antspk) [reply README.md:348]: Good idea, I'll update it.

**Review by rluvaton — DISMISSED** [review]
Great job!

I still want @goldbergyoni opinion so I'm gonna request his review too

File: README.md [hunk 4]
```diff
@@ -338,24 +338,41 @@ const count = 2 // it tries to run 2(), but 2 is not a function
 
 ## ![✔] 3.6 Use naming conventions for variables, constants, functions and classes
 
-**TL;DR:** Use **_lowerCamelCase_** when naming constants, variables and functions and **_UpperCamelCase_** (capital first letter as well) when naming classes. This will help you to easily distinguish between plain variables/functions, and classes that require instantiation. Use descriptive names, but try to keep them short
+**TL;DR:** Use **_lowerCamelCase_** when naming constants, variables and functions and **_UpperCamelCase_** (capital first letter as well) when naming classes, **_UPPER_SNAKE_CASE_** when naming global or static variables. This will help you to easily distinguish between plain variables/functions, classes that require instantiation and variables declared at global module scope. Use descriptive names, but try to keep them short
```
Comment (elektronik2k5) [inline README.md:341]:
```suggestion
**TL;DR:** Use **_lowerCamelCase_** when naming constants, variables and functions, **_UpperCamelCase_** (capital first letter as well) when naming classes and **_UPPER_SNAKE_CASE_** when naming global or static variables. This will help you to easily distinguish between plain variables, functions, classes that require instantiation and variables declared at global module scope. Use descriptive names, but try to keep them short
```

File: README.md [hunk 5]
```diff
@@ -338,24 +338,41 @@ const count = 2 // it tries to run 2(), but 2 is not a function
...
 };
 
-// for variables and functions names we use lowerCamelCase
-let someVariableExample = "value";
-function doSomething() {}
+// examples of UPPER_SNAKE_CASE convetion in nodejs/javascript ecosystem
+// in javascript Math.PI module
+const PI = 3.141592653589793;
+
+// https://github.com/nodejs/node/blob/master/lib/internal/http2/core.js
```
Comment (elektronik2k5) [inline README.md:359]: 1. Permalink, in case the file is renamed/moved/deleted
2. Includes line number
```suggestion
// https://github.com/nodejs/node/blob/b9f36062d7b5c5039498e98d2f2c180dca2a7065/lib/internal/http2/core.js#L303
```

File: README.md [hunk 6]
```diff
@@ -338,24 +338,41 @@ const count = 2 // it tries to run 2(), but 2 is not a function
...
+class SomeClassExample {
+  // for static class properties we use UPPER_SNAKE_CASE
+  static STATIC_PROPERTY = "value";
+}
+
+// for functions names we use lowerCamelCase
+function doSomething() {
+  // for scoped variable names we use the const/let keyword and lowerCamelCase
+  const someConstExample = "immutable value";
+  let someVariableExample = "mutable value";
```
Comment (elektronik2k5) [inline README.md:374]:
```suggestion
  let someVariableExample = "mutable value";
```
```suggestion
  let someMutableExample = "mutable value";
```

**Review by elektronik2k5 — CHANGES_REQUESTED** [review]

**Review by elektronik2k5 — APPROVED** [review]
Thank you!

**Review by rluvaton — APPROVED** [review]

**Comment by rluvaton** [comment]
Great job!

**Review by js-kyle — APPROVED** [review]

**Comment by goldbergyoni** [comment]
@antspk I was sure that this PR was merged. Suddenly noticed that it is not. My apologies, just merged.

I find this work, the discussions, and the final wording to be outstanding. We would love to see you participating in more work-streams here!

Many thanks.

@all-contributors please add @antspk for content

**Comment by goldbergyoni** [comment]
@antspk Minor note: In your final wording you did not use the 'screaming case' which I find to be more colorful and vibrant. M
